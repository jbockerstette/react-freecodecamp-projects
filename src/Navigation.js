import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.css';

export const ButtonLink = ({ className, bgColor, ...rest }) => {
  const cls = ['btn', className].join(' ');
  return <a className={cls} style={{ background: bgColor }} {...rest} />;
};
const navGridClass = ['nav-grid'];

const Navigation = ({ className }) => {
  navGridClass.push(className);
  const cls = navGridClass.join(' ');
  return (
    <div className={cls}>
      <NavLink
        className="nav-item"
        activeClassName="selected"
        to="/random-quote-machine"
      >
        Random Quote Machine
      </NavLink>
      <NavLink
        className="nav-item"
        activeClassName="selected"
        to="/markdown-previewer"
      >
        Markdown Previewer
      </NavLink>
      <NavLink
        className="nav-item"
        activeClassName="selected"
        to="/drum-machine"
      >
        Drum Machine
      </NavLink>
      <NavLink className="nav-item" activeClassName="selected" to="/calculator">
        Calculator
      </NavLink>
      <NavLink
        className="nav-item"
        activeClassName="selected"
        to="/pomodoro-clock"
      >
        Pomodoro Clock
      </NavLink>
      <ButtonLink
        className="fa fa-github nav-item"
        href="https://github.com/jbockerstette/react-freecodecamp-projects"
        target="_blank"
      />
    </div>
  );
};

export default Navigation;
