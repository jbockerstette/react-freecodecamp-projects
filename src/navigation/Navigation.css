.nav-grid {
  display: grid;
  grid-template-columns: repeat(5, 100px) 1fr;
  grid-template-rows: 64px;
  background: rgb(0, 95, 158);
}

.nav-item {
  display: flex;
  align-items: center;
  /* border: solid whitesmoke 1px; */
  color: whitesmoke;
  text-decoration: none;
  padding: 8px;
  height: 64px;
  align-self: center;
}

.nav-item:hover {
  opacity: 0.5;
}

.selected {
  font-weight: bold;
  background: rgb(0, 68, 114);
}
