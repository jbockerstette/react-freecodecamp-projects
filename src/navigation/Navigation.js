import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.css';
import routeConfig from '../routes/routeConfig';

export const ButtonLink = ({ className, bgColor, ...rest }) => {
  const cls = ['btn', className].join(' ');
  return (
    <a className={cls} style={{ background: bgColor }} {...rest}>
      {' '}
    </a>
  );
};
const navGridClass = ['nav-grid'];

const Navigation = ({ className }) => {
  navGridClass.push(className);
  const cls = navGridClass.join(' ');
  return (
    <div className={cls}>
      {routeConfig.map(route => (
        <NavLink
          key={route.path}
          className="nav-item"
          activeClassName="selected"
          to={route.path}
        >
          {route.title}
        </NavLink>
      ))}
      <ButtonLink
        className="fa fa-github fa-2x nav-item git-hub-link"
        href="https://github.com/jbockerstette/react-freecodecamp-projects"
        target="_blank"
      />
    </div>
  );
};

export default Navigation;
