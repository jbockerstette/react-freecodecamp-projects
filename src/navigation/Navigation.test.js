import React from 'react';
import renderer from 'react-test-renderer';
import { Simulate } from 'react-dom/test-utils';
import { MemoryRouter, Route } from 'react-router-dom';
import { mount } from 'enzyme';
import Navigation from './Navigation';

describe('Navigation', () => {
  // a way to render any part of your app inside a MemoryRouter
  // you pass it a list of steps to execute when the location
  // changes, it will call back to you with stuff like
  // `match` and `location`, and `history` so you can control
  // the flow and make assertions.
  const renderTestSequence = ({
    initialEntries,
    initialIndex,
    subject: Subject,
    steps
  }) => {
    const div = document.createElement('div');

    class Assert extends React.Component {
      componentDidMount() {
        this.assert();
      }

      componentDidUpdate() {
        this.assert();
      }

      assert() {
        const nextStep = steps.shift();
        if (nextStep) {
          nextStep({ ...this.props, div });
        } else {
          unmountComponentAtNode(div);
        }
      }

      render() {
        return this.props.children;
      }
    }

    const Test = () => (
      <MemoryRouter initialIndex={initialIndex} initialEntries={initialEntries}>
        <Route
          render={props => (
            <Assert {...props}>
              <Subject />
            </Assert>
          )}
        />
      </MemoryRouter>
    );

    mount(<Test />, { attachTo: div });
  };
  it('navigates around', done => {
    renderTestSequence({
      initialEntries: ['/random-quote-machine'],
      // tell it the subject you're testing
      subject: Navigation,

      // and the steps to execute each time the location changes
      steps: [
        // initial render
        ({ div, location }) => {
          // assert the screen says what we think it should
          expect(location.pathname).toEqual('/random-quote-machine');
          expect(div.querySelector('.nav-item.selected').text).toEqual(
            'Random Quote Machine'
          );

          // now we can imperatively navigate as the test
          Simulate.click(div.querySelector('[href="/markdown-previewer"]'), {
            button: 0
          });
        },

        // second render from new location
        ({ div, location }) => {
          expect(location.pathname).toEqual('/markdown-previewer');
          expect(div.querySelector('.nav-item.selected').text).toEqual(
            'Markdown Previewer'
          );

          // now we can imperatively navigate as the test
          Simulate.click(div.querySelector('[href="/drum-machine"]'), {
            button: 0
          });
        },

        // third render from new location
        ({ div, location }) => {
          expect(location.pathname).toEqual('/drum-machine');
          expect(div.querySelector('.nav-item.selected').text).toEqual(
            'Drum Machine'
          );

          // now we can imperatively navigate as the test
          Simulate.click(div.querySelector('[href="/calculator"]'), {
            button: 0
          });
        },

        ({ div, location }) => {
          expect(location.pathname).toEqual('/calculator');
          expect(div.querySelector('.nav-item.selected').text).toEqual(
            'Calculator'
          );

          // now we can imperatively navigate as the test
          Simulate.click(div.querySelector('[href="/pomodoro-clock"]'), {
            button: 0
          });
        },
        ({ div, location }) => {
          expect(location.pathname).toEqual('/pomodoro-clock');
          expect(div.querySelector('.nav-item.selected').text).toEqual(
            'Pomodoro Clock'
          );

          done();
        }
      ]
    });
  });
  it('should render correctly', () => {
    const comp = renderer.create(
      <MemoryRouter>
        <Navigation className="my-class-name" />
      </MemoryRouter>
    );
    const tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
