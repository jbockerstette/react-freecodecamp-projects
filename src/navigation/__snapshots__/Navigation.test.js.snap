// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
<div
  className="nav-grid       my-class-name"
>
  <a
    aria-current={null}
    className="nav-item"
    href="/random-quote-machine"
    onClick={[Function]}
    style={undefined}
  >
    Random Quote Machine
  </a>
  <a
    aria-current={null}
    className="nav-item"
    href="/markdown-previewer"
    onClick={[Function]}
    style={undefined}
  >
    Markdown Previewer
  </a>
  <a
    aria-current={null}
    className="nav-item"
    href="/drum-machine"
    onClick={[Function]}
    style={undefined}
  >
    Drum Machine
  </a>
  <a
    aria-current={null}
    className="nav-item"
    href="/calculator"
    onClick={[Function]}
    style={undefined}
  >
    Calculator
  </a>
  <a
    aria-current={null}
    className="nav-item"
    href="/pomodoro-clock"
    onClick={[Function]}
    style={undefined}
  >
    Pomodoro Clock
  </a>
  <a
    aria-current={null}
    className="nav-item"
    href="/survey-form"
    onClick={[Function]}
    style={undefined}
  >
    Survey Form
  </a>
  <a
    className="btn fa fa-github fa-2x nav-item git-hub-link"
    href="https://github.com/jbockerstette/react-freecodecamp-projects"
    style={
      Object {
        "background": undefined,
      }
    }
    target="_blank"
  >
     
  </a>
</div>
`;
